spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/query/PartTreeDatastoreQuery.java

```diff
@@ -81,12 +81,7 @@ public PartTreeDatastoreQuery(DatastoreQueryMethod queryMethod,
 	}
 
 	private void validateAndSetFilterParts() {
-		if (this.tree.isDelete()) {
-			throw new UnsupportedOperationException(
-					"Delete queries are not supported in Cloud Datastore: "
-							+ this.queryMethod.getName());
-		}
-		else if (this.tree.isDistinct()) {
+		if (this.tree.isDistinct()) {
 			throw new UnsupportedOperationException(
```

Comment:
I distinctly remember doing something to support this?

Reply:
I remember now. We decided to NOT support projection queries in part-tree because of how cumbersome it is to set up and use from the user's end. Without projection queries we can no longer set the distinct and it makes no sense because the Key is always selected and is always distinct.